src/main/java/software/amazon/lambda/snapstart/ByteCodeIntrospector.java

```diff
@@ -43,7 +43,29 @@ private static Set<String> setOf(String ... strings) {
     };
 
     boolean isLambdaHandler(XClass xClass) {
-        return implementsLambdaInterface(xClass) || hasLambdaHandlerMethod(xClass);
+        return implementsLambdaInterface(xClass) || 
+               hasLambdaHandlerMethod(xClass) || 
+               (hasHandlerInClassName(xClass) && hasMethodCalledHandleRequest(xClass));
+    }
+
+    /**
+     * Returns true if the class has the word "handler" in the name.
+     */
+    private boolean hasHandlerInClassName(XClass xClass) {
+        return xClass.toString().contains("Handler");
+    }
+
+    /**
+     * Returns true if the class has a method called "handleRequest"
+     */
+    private boolean hasMethodCalledHandleRequest(XClass xClass) {
```

Comment:
Got it. Thanks

Reply:
You didn't update this naming yet. We can discuss if you prefer to not change the name.

Reply:
You're right. I first thought it might be confusing since we already have a `hasLambdaHandlerMethod` method. That's why I tried to make it more explicit and remove any confusion by naming it the way I did. But I think the method's comment is clear enough in this regard, so `hasHandleRequestMethod` makes more sense.